tests/daemon_test_fixture.cpp

```diff
@@ -525,29 +525,6 @@ bool mpt::DaemonTestFixture::is_ready(std::future<R> const& f)
     return f.wait_for(std::chrono::seconds(5)) == std::future_status::ready;
 }
 
-template <typename DaemonSlotPtr, typename Request, typename Server>
-grpc::Status mpt::DaemonTestFixture::call_daemon_slot(Daemon& daemon, DaemonSlotPtr slot, const Request& request,
-                                                      Server&& server)
-{
-    std::promise<grpc::Status> status_promise;
-    auto status_future = status_promise.get_future();
-
-    auto thread = QThread::create([&daemon, slot, &request, &server, &status_promise] {
-        QEventLoop loop;
-        (daemon.*slot)(&request, &server, &status_promise);
-        loop.exec();
-    });
-
-    thread->start();
-
-    EXPECT_TRUE(is_ready(status_future));
-
-    thread->quit();
-    thread->wait();
-
-    return status_future.get();
-}
-
```

Comment:
Why the move to the header?  There are the actual templates still defined below.  This was *supposed* to help with compiling *only* for the types we need, not every single type that could possibly be.

Reply:
Hi Chris, I kept wondering how this could have worked before, I totally missed those declarations below. The template would still only be instantiated for the types we used it with, but that could happen multiple times for the same types, which the current scheme avoids. I'll revert and adhere to it as well instead.

Reply:
Right, thanks!